File: server/scripts/createUser.js
import "dotenv/config";
import { createUser, initDb, ROLE_ADMIN, ROLE_AFFILIATE, ROLE_USER } from "../db.js";

const parseArgs = () => {
  const args = process.argv.slice(2);
  const map = {};
  for (let i = 0; i < args.length; i += 1) {
    const key = args[i];
    if (!key.startsWith("--")) continue;
    map[key.replace("--", "")] = args[i + 1];
    i += 1;
  }
  return map;
};

const main = async () => {
  const args = parseArgs();
  const username = args.username;
  const password = args.password;
  const role = args.role ?? ROLE_USER;
  const credits = Number.parseInt(args.credits ?? "0", 10) || 0;

  if (!username || !password) {
    console.error("Usage: node server/scripts/createUser.js --username <email> --password <senha> [--role admin|user|affiliate] [--credits 10]");
    process.exit(1);
  }

  if (![ROLE_ADMIN, ROLE_USER, ROLE_AFFILIATE].includes(role)) {
    console.error("Role invalida. Use admin, user ou affiliate.");
    process.exit(1);
  }

  await initDb();
  const user = await createUser({ username, password, role, credits });

  console.log("User created:");
  console.log(`  id: ${user.id}`);
  console.log(`  username: ${user.username}`);
  console.log(`  role: ${user.role}`);
  console.log(`  credits: ${user.credits}`);
  console.log(`  api_key: ${user.api_key}`);
};

main().catch((error) => {
  console.error(error.message || error);
  process.exit(1);
});

File: backend/scripts/create_user.py
from __future__ import annotations

import argparse
from pathlib import Path
import sys

BASE_DIR = Path(__file__).resolve().parents[1]
sys.path.append(str(BASE_DIR))

from app.config import settings  # noqa: E402
from app.db import ROLE_ADMIN, ROLE_AFFILIATE, ROLE_USER, create_user, init_db  # noqa: E402


def main() -> None:
    parser = argparse.ArgumentParser(description="Create a user in the MEGA FACIL database")
    parser.add_argument("--username", required=True)
    parser.add_argument("--password", required=True)
    parser.add_argument("--role", default=ROLE_USER, choices=[ROLE_ADMIN, ROLE_USER, ROLE_AFFILIATE])
    parser.add_argument("--credits", type=int, default=0)
    parser.add_argument(
        "--db-path",
        default=str(settings.db_path),
        help="Path to SQLite database file",
    )

    args = parser.parse_args()
    db_path = Path(args.db_path)

    init_db(db_path)
    user = create_user(db_path, args.username, args.password, args.role, args.credits)

    print("User created:")
    print(f"  id: {user.id}")
    print(f"  username: {user.username}")
    print(f"  role: {user.role}")
    print(f"  credits: {user.credits}")
    print(f"  api_key: {user.api_key}")


if __name__ == "__main__":
    main()
